cmd/nerdctl/compose_port_linux_test.go

```diff
@@ -0,0 +1,61 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/containerd/nerdctl/pkg/testutil"
+)
+
+func TestComposePort(t *testing.T) {
+	// when no port mapping is found, docker compose v1 prints `\n` while v2 prints `:0\n`
+	// both v1 and v2 have exit code 0 (succeess)
+	// nerdctl compose will fail with error (no public port fouond).
+	testutil.DockerIncompatible(t)
```

Comment:
Can we split the incompatible part to another test?

Reply:
fixed.